source/family/cypress/target_reset.c

```diff
@@ -0,0 +1,154 @@
+/*******************************************************************************
+* @file    target_reset.c
+* @brief   Target reset for the PSoC6 target
+*
+********************************************************************************
+* Copyright (2019) Cypress Semiconductor Corporation
+* or a subsidiary of Cypress Semiconductor Corporation.
+*
+* Licensed under the Apache License, Version 2.0 (the "License"); you may
+* not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+* WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*****************************************************************************/
+
+#include "target_reset.h"
+#include "swd_host.h"
+#include "swd.h"
+#include "cyfitter.h"
+#include "PSOC6xxx.h"
+#include "target_config.h"
+#include "target_family.h"
+#include "IO_Config.h"
+
+#define MAX_AP            (2u)
+#define CM4_AP            (2u << 24u)
+#define CPU_ID            (0xE000ED00u)
+#define READY_ADDR        (0x4023004Cu)
+#define CPUSS_CM4_PWR_CTL (0x40210080u)
+#define READY_VAL_SEQ     (0x12344321u)
+
+extern uint32_t DAP_AP;
+
+bool isTargetAcquired;
+extern uint32_t SWDAcquirePSoC6BLE(uint8_t acquireMode, uint8_t attempts);
+extern uint8_t SWD_SelectAP(uint32_t ap);
+extern uint8_t SWD_WriteCSW(uint32_t csw_val);
+extern uint8_t SWD_Send(uint32_t addr_32, uint32_t  data_32);
+extern uint8_t SWD_Receive(uint32_t adr_32, uint32_t *data_32);
+extern uint8_t SWD_Abort(void);
+
+void target_before_init_debug(void)
+{
+    // any target specific sequences needed before attaching
+    //	to the DAP across JTAG or SWD
+	  if (!isTargetAcquired)
+		{
+			uint32_t status = SWDAcquirePSoC6BLE(0, 1);
+			if (kit_has_sflash_restriction())
+			{
+				uint32_t ready_val = 0;
+				for ( uint32_t i = 0; i < 6000u; i++ )
+				{
+					ready_val = 0;
+					if ((SWD_Receive(READY_ADDR, &ready_val) == SWD_ACK) && (ready_val == READY_VAL_SEQ))
+					{
+						isTargetAcquired = true;
+						break;
+					}
+					CyDelay(10u);
+				}
+			}
+			else
+			{
+				if (status == ACQUIRE_PASS)
+				{
+					isTargetAcquired = true;
+				}
+			}
+			
+			if (!isTargetAcquired)
+			{
+				return;
+			}
+			
+			// find valid AP with core
+			uint32_t cpu_id = 0;
+			for ( uint32_t ap_index = 0; cpu_id == 0 && ap_index <= MAX_AP ; ap_index++) 
+			{
+				DAP_AP = ap_index << 24;
+				if ( (SWD_SelectAP(DAP_AP) != SWD_ACK) || (SWD_Receive(CPU_ID, &cpu_id) != SWD_ACK) )
```

Comment:
Please use common SWD implementation here.

Reply:
I switched to your proposed implementation.